bob/src/api/http.rs

```diff
@@ -633,6 +643,84 @@ fn internal(message: String) -> StatusExt {
     StatusExt::new(Status::InternalServerError, false, message)
 }
 
+fn parse_error(message: impl Into<String>) -> StatusExt {
+    StatusExt::new(Status::BadRequest, false, message.into())
+}
+
+impl DataKey {
+    fn from_bytes(bytes: Vec<u8>) -> Result<Self, StatusExt> {
+        let mut key = [0u8; BOB_KEY_SIZE];
+        key.iter_mut()
+            .rev()
+            .zip(bytes.iter().rev())
+            .for_each(|(a, b)| {
+                *a = *b;
+            });
+        if bytes.len() > BOB_KEY_SIZE {
```

Comment:
I think you should check length before `key` initialization above

Reply:
Done